packages/react-theme-provider/src/ThemeProvider.styles.ts

```diff
@@ -0,0 +1,27 @@
+import { memoizeFunction } from '@uifabric/utilities';
+import { mergeStyles } from '@uifabric/styling';
+import { TokenSetType } from './types';
+
+export const getBodyStyles = memoizeFunction((bodyTheme: TokenSetType | undefined): string | undefined => {
```

Comment:
This could be `makeStyles`:

```tsx
export const useBodyStyles = makeStyles(theme => {
  ...
});
```

Would that work?

Reply:
useThemeProviderStyles

Reply:
I'm wondering if we actually need to spit out the entire set of theme tokens in this class. Basically we are now providing some tokens as inline styles and some as a class. Maybe we should simply try moving them all into makeStyles.

Also, the other thing I'm doing: in a separately PR I'll change makeStyles to use theme.renderRules to render rules in any way we want, which will allow us to flip between mergeStyles, jss, emotion, or whatever tool we need. this will help with convergence and make us less dependent on one css solution.